source/extensions/omni.isaac.orbit/omni/isaac/orbit/sim/utils.py

```diff
@@ -777,3 +777,47 @@ def find_global_fixed_joint_prim(
                 break
 
     return fixed_joint_prim
+
+
+"""
+USD Variants.
+"""
+
+
+def set_usd_variants(prim_path: str, variants: object | dict[str, str], stage: Usd.Stage | None = None) -> None:
+    """
+    Sets the variant selections for the specified variant sets on a USD prim.
+
+    Args:
+        prim_path (str): The path of the USD prim.
+        variants (dict[str, str]): A dictionary mapping variant set names to variant selections.
+        stage (Usd.Stage | None, optional): The USD stage. If not provided, the current stage will be used.
```

Comment:
We maintain type-hinting only at annotation-level (not in the docstrings)

```suggestion
        prim_path: The path of the USD prim.
        variants: A dictionary mapping variant set names to variant selections.
        stage: The USD stage. Defaults to None, in which case, the current stage is used.
```